demo/agw/ScrolledThumbnail.py

```diff
@@ -0,0 +1,640 @@
+        self.radiostyle3 = wx.RadioButton(self.panel, -1, "THUMB_OUTLINE_RECT")
+        self.radiostyle4 = wx.RadioButton(self.panel, -1, "THUMB_OUTLINE_IMAGE")
+        self.highlight = wx.CheckBox(self.panel, -1, "Highlight on pointing")
+        self.showfiles = wx.CheckBox(self.panel, -1, "Show file names")
+        self.enabledragging = wx.CheckBox(self.panel, -1, "Enable drag and drop")
+        self.setpopup = wx.CheckBox(self.panel, -1, "Set popup menu on thumbs")
+        self.setgpopup = wx.CheckBox(self.panel, -1, "Set global popup menu")
+        self.enabletooltip = wx.CheckBox(self.panel, -1, "Enable thumb tooltips")
+        self.textzoom = wx.TextCtrl(self.panel, -1, "1.4")
+        self.zoombutton = wx.Button(self.panel, -1, "Set zoom factor")
+        self.textthumbwidth = wx.TextCtrl(self.panel, -1, "96")
+        self.textthumbheight = wx.TextCtrl(self.panel, -1, "80")
+        self.thumbsizebutton = wx.Button(self.panel, -1, "Set thumbnail size (WxH)")
+        self.fontbutton = wx.Button(self.panel, -1, "Set caption font")
+        self.colourbutton = wx.Button(self.panel, -1, "Set selection colour")
+
+        self.radios = [self.radiostyle1, self.radiostyle2, self.radiostyle3,
+                       self.radiostyle4]
+        self.thumbstyles = ["THUMB_OUTLINE_NONE", "THUMB_OUTLINE_FULL", "THUMB_OUTLINE_RECT",
+                            "THUMB_OUTLINE_IMAGE"]
+
+        self.SetProperties()
+        self.DoLayout()
+
+        self.panel.SetSizer(sizer)
+        sizer.Layout()
+
+        self.Bind(wx.EVT_RADIOBUTTON, self.OnChangeOutline, self.radiostyle1)
+        self.Bind(wx.EVT_RADIOBUTTON, self.OnChangeOutline, self.radiostyle2)
+        self.Bind(wx.EVT_RADIOBUTTON, self.OnChangeOutline, self.radiostyle3)
+        self.Bind(wx.EVT_RADIOBUTTON, self.OnChangeOutline, self.radiostyle4)
+        self.Bind(wx.EVT_CHECKBOX, self.OnHighlight, self.highlight)
+        self.Bind(wx.EVT_CHECKBOX, self.OnShowFiles, self.showfiles)
+        self.Bind(wx.EVT_CHECKBOX, self.OnEnableDragging, self.enabledragging)
+        self.Bind(wx.EVT_CHECKBOX, self.OnSetPopup, self.setpopup)
+        self.Bind(wx.EVT_CHECKBOX, self.OnSetGlobalPopup, self.setgpopup)
+        self.Bind(wx.EVT_CHECKBOX, self.OnEnableToolTips, self.enabletooltip)
+        self.Bind(wx.EVT_BUTTON, self.OnSetZoom, self.zoombutton)
+        self.Bind(wx.EVT_BUTTON, self.OnSetThumbSize, self.thumbsizebutton)
+        self.Bind(wx.EVT_BUTTON, self.OnSetFont, self.fontbutton)
+        self.Bind(wx.EVT_BUTTON, self.OnSetColour, self.colourbutton)
+        self.Bind(wx.EVT_BUTTON, self.OnSetDirectory, self.dirbutton)
+
+        self.scroll.Bind(EVT_THUMBNAILS_SEL_CHANGED, self.OnSelChanged)
+        self.scroll.Bind(EVT_THUMBNAILS_POINTED, self.OnPointed)
+        self.scroll.Bind(EVT_THUMBNAILS_DCLICK, self.OnDClick)
+
+        splitter.SplitVertically(self.scroll, self.panel, 400)
+
+        splitter.SetMinimumPaneSize(140)
+        self.SetMinSize((1000, 1000))
```

Comment:
If you don't also set the frame's size here then it's possible to see the frame at its small initial size and then resize to the minsize set here.

Also, (1000,1000) seems way to big for this content. Once you fix the sizer-related things above, and if you give the frame a sizer containing the splitter, then you may be able to have the frame size itself with `self.Fit()`

Reply:
Using wx.GetDisplaySixe() might help. I do this intelligently in my programs, but a small bit of code may be needed to do this accurately. If in doubt, use Size(800, 600) by default. Obiously rasp pi would be smaller, but this is primarily targeted towards desktops still at the moment.
A ALL-EMCOMPASSING function would mean lots of writing plus performance differences.

Reply:
At Size(800,600) the options at the bottom of the control panel are cut off.  I set the Frame size to (850,800) in the constructor and eliminated the SetMinSize() call.

I tried to use self.Fit() but that results in a very small frame (maybe 50x50).